src/edu/washington/escience/myria/operator/agg/AggUtils.java

```diff
@@ -84,14 +88,19 @@ public static boolean needsStats(final Set<AggregationOp> aggOps) {
    *
    * @param factories The factories that will produce the aggregators.
    * @param inputSchema The schema of the input tuples.
+   * @param pyFuncRegistrar - python function registrar to get functions from workers.
    * @return the aggregators for this operator.
    * @throws DbException if there is an error.
    */
   public static Aggregator[] allocateAggs(
-      final AggregatorFactory[] factories, final Schema inputSchema) throws DbException {
+      final AggregatorFactory[] factories,
+      final Schema inputSchema,
+      final PythonFunctionRegistrar pyFuncRegistrar)
```

Comment:
It is required by all user-defined-aggregates, not needed by countall or single column aggs.
which means I need to have a special case for userdefinedagg factory constructor --  which makes the interface non-uniform.  It is not the perfect design,  I am open to discussion of how aggregators could be refactored in the future.